examples/esm-http-ts/index.ts

```diff
@@ -0,0 +1,43 @@
+import { registerInstrumentations } from '@opentelemetry/instrumentation';
+import { trace, DiagConsoleLogger, DiagLogLevel, diag } from '@opentelemetry/api';
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http';
+import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
+import {
+  ConsoleSpanExporter,
+  SimpleSpanProcessor,
+} from '@opentelemetry/sdk-trace-base';
+import { Resource } from '@opentelemetry/resources';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import http from 'http';
```

Comment:
is ESM support/loader really needed for this sample? There is actually no ESM module included.

Or is RITM also bypassed for built ins which are more like CJS as of now.

Reply:
The example app is built in TypeScript and compiles to ESM in JavaScript. Without this change, if we just compiled to CJS it does not work. This example is intended to demonstrate how to configure an ESM app to work with auto-instrumentation.